pkg/clientinfo/diagnostics.go

```diff
@@ -118,6 +124,32 @@ func (r *Registry) RegisterClientInfoSource(
 	})
 }
 
+// RegisterChainInfo registers the diagnostics source providing
+// information about chains.
+func (r *Registry) RegisterChainInfo(
+	blockCounter chain.BlockCounter,
+) {
+	r.RegisterDiagnosticSource("chain_info", func() string {
+		currentBlock, err := blockCounter.CurrentBlock()
+
+		if err != nil {
+			return ""
+		}
+
+		chainInfo := Chain{
+			EthBlockNumber: currentBlock,
+		}
+
+		bytes, err := json.Marshal(chainInfo)
+		if err != nil {
+			logger.Error("error on serializing chain info to JSON: [%v]", err)
```

Comment:
Shouldn't this be `Errorf`?

Reply:
Replaced with Errorf [0493b72](https://github.com/keep-network/keep-core/pull/3637/commits/0493b72b3f8e4c41684037eaa534beef7433474c)